scripts/cl_acquisition.py

```diff
@@ -60,120 +66,268 @@ def __init__(self):
+          index and content, because X should be scanned first, so it's last
+          dimension in the index.
+        """
+        rep = self.repetition
+        roi = self.roi
+        width = (roi[2] - roi[0], roi[3] - roi[1])
 
-    def set_roi(self, rect):
-        # save the resolution
-        res = self.escan.resolution.value
-        center = ((rect[0] + rect[2]) / 2, (rect[1] + rect[3]) / 2)
-        width = (rect[2] - rect[0], rect[3] - rect[1])
+        # Take into account the "border" around each pixel
+        pxs = (width[0] / rep[0], width[1] / rep[1])
+        lim = (roi[0] + pxs[0] / 2, roi[1] + pxs[1] / 2,
+               roi[2] - pxs[0] / 2, roi[3] - pxs[1] / 2)
 
         shape = self.escan.shape
-        scale = (1 / width[0], 1 / width[1])
-        # translation is distance from center (situated at 0.5, 0.5), can be floats
-        trans = (shape[0] * (center[0] - 0.5), shape[1] * (center[1] - 0.5))
-        # always in this order
-        self.escan.scale.value = scale # might update the resolution & translation
-        self.escan.resolution.value = res
-        self.escan.translation.value = trans
+        # convert into SEM translation coordinates: distance in px from center
+        # (situated at 0.5, 0.5), can be floats
+        lim_main = (shape[0] * (lim[0] - 0.5), shape[1] * (lim[1] - 0.5),
+                    shape[0] * (lim[2] - 0.5), shape[1] * (lim[3] - 0.5))
+        logging.info("Generating points in the SEM area %s, from rep %s and roi %s",
+                     lim_main, rep, roi)
 
-    def acquire_cube(self):
+        pos = numpy.empty((rep[1], rep[0], 2), dtype=numpy.float)
+        posy = pos[:,:, 1].swapaxes(0, 1)  # just a view to have Y as last dim
+        posy[:,:] = numpy.linspace(lim_main[1], lim_main[3], rep[1])
+        # fill the X dimension
+        pos[:,:, 0] = numpy.linspace(lim_main[0], lim_main[2], rep[0])
+
+        return pos
+
+    def get_pixel_size(self) -> Tuple[float, float]:
+        """
+        Computes the pixel size (based on the repetition, roi and FoV of the
+          e-beam). The RepetitionStream does provide a .pixelSize VA, which
+          should contain the same value, but that VA is for use by the GUI.
+        return (float, float): pixel size in m.
+        """
+        epxs = self.escan.pixelSize.value
+        rep = self.repetition
+        roi = self.roi
+        eshape = self.escan.shape
+
+        phy_size_x = (roi[2] - roi[0]) * epxs[0] * eshape[0]
+        phy_size_y = (roi[3] - roi[1]) * epxs[1] * eshape[1]
+        pxsx = phy_size_x / rep[0]
+        pxsy = phy_size_y / rep[1]
+        logging.debug("px size guessed = %s x %s", pxsx, pxsy)
+
+        return (pxsx, pxsy)
+
+    def get_center_pxs(self, datatl):
+        """
+        Computes the center and pixel size of the entire data based on the
+        top-left data acquired.
+        datatl (DataArray): first data array acquired
```

Comment:
Function parameter consistency according to coding-guidelines.rst:
```suggestion
        :param datatl (DataArray): first data array acquired
```